src/server/pegasus_server_impl_init.cpp

```diff
@@ -527,6 +527,12 @@ pegasus_server_impl::pegasus_server_impl(dsn::replication::replica *r)
     _key_ttl_compaction_filter_factory = std::make_shared<KeyWithTTLCompactionFilterFactory>();
     _data_cf_opts.compaction_filter_factory = _key_ttl_compaction_filter_factory;
 
+    _data_cf_opts.periodic_compaction_seconds =
+        dsn_config_get_value_uint64("pegasus.server",
+                                    "rocksdb_periodic_compaction_seconds",
+                                    30 * 24 * 60 * 60,
```

Comment:
I think the default value should be UINT64_MAX-1, which is default value in rocksdb

Reply:
The default value of periodic_compaction_seconds is 30 days.It is turned on in column_family.
Below is code of rocksdb.
```
  const uint64_t kAdjustedPeriodicCompSecs = 30 * 24 * 60 * 60;

  // Turn on periodic compactions and set them to occur once every 30 days if
  // compaction filters are used and periodic_compaction_seconds is set to the
  // default value.
  if (result.compaction_style != kCompactionStyleFIFO) {
    if ((result.compaction_filter != nullptr ||
         result.compaction_filter_factory != nullptr) &&
        result.periodic_compaction_seconds == kDefaultPeriodicCompSecs &&
        is_block_based_table) {
      result.periodic_compaction_seconds = kAdjustedPeriodicCompSecs;
    }
  }
```

Reply:
In my opinion, the default value is  kDefaultPeriodicCompSecs(UINT64_MAX -1). And rocksdb set it to 30days when you use default value. maybe someday rocksdb will change it to 20days or 10days, that would lead to inconsistencies

Reply:
@levy5307 the default value has been changed to 0